eduNEXT/frontend-app-validation-panel
# feat: implement POST endpoints

## Description
This PR implements the possibility of updating the validation process statuses to move the processes among the different stages in the validation.

## How to test
1. Make sure you have the repository cloned and accessed in the current branch of this PR ([bc/DS-641/connect-all-endpoints](https://github.com/eduNEXT/frontend-app-validation-panel/tree/bc/DS-641/connect-all-endpoints))
2. Make sure you have the Plugin CVW in the LMS
 - - You have to have some categories and validation bodies created in your DDBB in the [Platform Global Teacher Campus](http://local.overhang.io:8000/admin/platform_global_teacher_campus/) and have at least a course to execute the tests.
 - - You'll be a ```courseAuthor``` if your user is part of the Staff (http://local.overhang.io:8000/courses/course_id/instructor#view-membership) of the course and you will be a ```validator``` if your user belongs to a validation body (http://local.overhang.io:8000/admin/platform_global_teacher_campus/validationbody/)

3. Override the ```course-authoring``` MFE to run the current changes:
```yml
version: '3.7'
services:
  course-authoring:
    volumes:
      - 'path/to/frontend-app-validation-panel:/openedx/app'
```
4. Now you should be able of accessing to this path http://apps.local.overhang.io:2001/validation-panel 
5. Execute the following tasks:
- Course author
  - You can submit a course for validation by clicking the button for it, filling in the required info, and clicking on the submit button
![image](https://github.com/eduNEXT/frontend-app-validation-panel/assets/86393372/6018fe2a-208a-4adf-afa0-7b568f5c7aea)
  - You can cancel a validation process by clicking the ```Cancel validation``` button
![image](https://github.com/eduNEXT/frontend-app-validation-panel/assets/86393372/830f438d-ab3f-41e1-9f68-27cdba4c3ba3)
  - You can re-submit the validation process by clicking the ```Re-submit for validation``` button
![image](https://github.com/eduNEXT/frontend-app-validation-panel/assets/86393372/9baaeae9-e18a-4c1e-bdad-14eb990b9149)

- Validator
  -  You can select a course for reviewing it by clicking the ```Review course``` button
![image](https://github.com/eduNEXT/frontend-app-validation-panel/assets/86393372/4c182d6c-c506-4f5f-932b-0cc6fa59337d)
  - You can release it and allow another validator to do it by clicking the ```Release course``` button or, if you want to submit a review, click on the title and you'll be accessed to a modal for doing it
![image](https://github.com/eduNEXT/frontend-app-validation-panel/assets/86393372/3c2f4261-ccad-4a54-90c5-1e80c0ad8b80)

  - This is the modal for submitting the review, you just have to fill in the information and submit it
![image](https://github.com/eduNEXT/frontend-app-validation-panel/assets/86393372/932574a3-7338-4569-b1ec-c41591cc48a5)

**The validation process will be updated for the course author too:**
  
![image](https://github.com/eduNEXT/frontend-app-validation-panel/assets/86393372/2056eac4-e451-4f8d-973f-35bd6bc5669e)

## Additional
If as validator during the review you try to set the status to 'Disapproved', the field 'Reason' will be enabled, but at this moment the select does not count with any option because is not connected to the get endpoint yet; that will be another PR
![image](https://github.com/eduNEXT/frontend-app-validation-panel/assets/86393372/b792d9ed-8cb6-4b81-bb3c-497bb0e84870)
